autobahn/wamp/cryptosign.py

```diff
@@ -354,7 +354,7 @@ def _verify_signify_ed25519_signature(pubkey_file, signature_file, message):
 
 if HAS_CRYPTOSIGN:
 
-    def format_challenge(session, challenge, channel_id_type) -> bytes:
+    def format_challenge(challenge, channel_id_raw, channel_id_type) -> bytes:
```

Comment:
@oberstet it might make sense to change this method to support `challenge` as a hex string as well.

One reason for that is that the worker would need to cbor serialize the challenge object manually and then the controller would need to deserialize that.

Reply:
my view: this is JS style;) we should add py3 type annotations (I did that for parts that I changed since this feature is available in functions that I touched .. partially at least), and also avoid `Optional`. in general, I would like to move the AB&CB codebases to type hinting inside. this has just so many advantages. now, of course, you could argue: why not `Optional` with str and bytes possible? well yes .. possible ... so I'd be fine with both these options. with type hinting

Reply:
Didn't need to change the above signature, as I went with a different solution in Crossbar. https://github.com/crossbario/crossbar/pull/1906/files#r752120365

Reply:
Docstring and annotations are added now